scripts/validate_docstrings.py

```diff
@@ -402,6 +402,12 @@ def examples_errors(self):
             error_msgs += f.getvalue()
         return error_msgs
 
+    @property
+    def examples_source_code(self):
+        codes = doctest.DocTestParser().get_examples(self.raw_doc)
+        codes = [line.source for line in codes]
+        return codes
```

Comment:
You can also return in the previous line.

May be `lines` is a better name? codes sounds to me like identifiers